.github/workflows/release-workflow.yml

```diff
@@ -0,0 +1,120 @@
+name: Release workflow
+# This workflow is triggered on creating tags to master or a opendistro release branch
+on:
+  #  push:
+  #    tags:
+  #      - 'v*'
+  push:
+    branches:
+      - workflows
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        java: [14]
+    # Job name
+    name: Build Asynchronous Search with JDK ${{ matrix.java }}
+    # This job runs on Linux
+    runs-on: ubuntu-latest
+    steps:
+      # This step uses the checkout Github action: https://github.com/actions/checkout
+      - name: Checkout Branch
+        uses: actions/checkout@v2
+      # This step uses the setup-java Github action: https://github.com/actions/setup-java
+      - name: Set Up JDK ${{ matrix.java }}
+        uses: actions/setup-java@v1
+        with:
+          java-version: ${{ matrix.java }}
+
+      # Building zip, deb and rpm files
+      - name: Build with Gradle
+        run: ./gradlew build buildDeb buildRpm --no-daemon -Dbuild.snapshot=false
+
+      # Creating artifact path as well as individual folders for rpm, zip and deb assets
+      - name: Create Artifact Path
+        run: |
+          mkdir -p asynchronous-search-artifacts
+          cp ./build/distributions/*.deb asynchronous-search-artifacts
+          cp ./build/distributions/*.deb asynchronous-search-artifacts_deb
+          cp ./build/distributions/*.rpm asynchronous-search-artifacts
+          cp ./build/distributions/*.rpm asynchronous-search-artifacts_rpm
+          cp ./build/distributions/*.zip asynchronous-search-artifacts
+          cp ./build/distributions/*.zip asynchronous-search-artifacts_zip
```

Comment:
whats the standard artifact naming convention?

Reply:
can we extract out the common?

Reply:
> whats the standard artifact naming convention?

referring to https://quip-amazon.com/gmuHA2uDtWHV/ODFE-artifact-naming-convention

From release workflow run logs : 
s3://staging.artifacts.opendistroforelasticsearch.amazon.com/snapshots/elasticsearch-plugins/asynchronous-search/opendistro-asynchronous-search-1.11.1.0-build-18.zip
s3://staging.artifacts.opendistroforelasticsearch.amazon.com/snapshots/elasticsearch-plugins/asynchronous-search/opendistro-asynchronous-search-1.11.1.0-build-18.deb
s3://staging.artifacts.opendistroforelasticsearch.amazon.com/snapshots/elasticsearch-plugins/asynchronous-search/opendistro-asynchronous-search-1.11.1.0-build-18.rpm

Sreekar from infra team okayed the names. Had shared an earlier build with him.